Simplenote/src/androidTest/java/com/automattic/simplenote/smoke/test/TestRunner.java

```diff
@@ -0,0 +1,405 @@
+            new IntroPage()
+                    .goToLoginWithEmail()
+                    .login(email, password);
+
+            new MainPage()
+                    .isOpened()
+                    .logout(email);
+        }
+    }
+
+    @Test
+    public void testLoginWithWrongPassword() {
+        TestUtils.logoutIfNecessary(email);
+
+        new IntroPage()
+                .goToLoginWithEmail()
+                .login(DataProvider.LOGIN_EMAIL, DataProvider.LOGIN_WRONG_PASSWORD);
+
+        new LoginPage()
+                .checkLoginFailedMessage();
+    }
+
+    @Test
+    public void testSearchingInTheSearchFieldDoesAGlobalSearch() {
+        NoteDTO noteDTO = DataProvider.generateNotes(1).get(0);
+
+        String searchParam = noteDTO.getTitle();
+
+        new IntroPage()
+                .goToLoginWithEmail()
+                .login(email, password);
+
+        new MainPage()
+                .switchCondensedNoteListMode(false)
+                .pressBack();
+
+        new MainPage()
+                .addNewNote(noteDTO)
+                .openSearchPage()
+                .search(searchParam)
+                .checkSearchResultsTitleAndContent(searchParam);
+
+        new MainPage()
+                .openNote(noteDTO)
+                .trash()
+                .logout(email);
+    }
+
+    @Test
+    public void testFilterByTagWhenClickingOnTagInTagDrawer() {
+        NoteDTO noteDTO = DataProvider.generateNotesWithUniqueContent(1).get(0);
+
+        new IntroPage()
+                .goToLoginWithEmail()
+                .login(email, password);
+
+        new MainPage()
+                .switchCondensedNoteListMode(false)
+                .pressBack();
+
+        new MainPage()
+                .addNewNote(noteDTO)
+                .selectTagFromDrawer(noteDTO.getTags().get(0))
+                .checkNoteContentIsInTheList(noteDTO.getContent().substring(0, 15));
+
+        new MainPage()
+                .openNote(noteDTO)
+                .trash()
+                .logout(email);
+    }
+
+    @Test
+    public void testViewTrashedNotesByClickingOnTrash() {
+        NoteDTO noteDTO = DataProvider.generateNotesWithUniqueContent(1).get(0);
+
+        new IntroPage()
+                .goToLoginWithEmail()
+                .login(email, password);
+
+        new MainPage()
+                .addNewNote(noteDTO)
+                .openNote(noteDTO)
+                .trash()
+                .openTrashPage()
+                .checkNoteIsTrashed(noteDTO.getTitle());
+
+        new MainPage()
+                .logout(email);
+    }
+
+    @Test
+    public void testRestoreNoteFromTrashScreen() {
+        NoteDTO noteDTO = DataProvider.generateNotesWithUniqueContent(1).get(0);
+
+        new IntroPage()
+                .goToLoginWithEmail()
+                .login(email, password);
+
+        new MainPage()
+                .addNewNote(noteDTO)
+                .openNote(noteDTO)
+                .trash()
```

Comment:
Up through the `trash()` step, this is the same sequence of steps as the `testTrashNote()` test. Is it worth combining tests like these that build on each other?

Reply:
This topic is related to `testLogin()` covering logout functionality. If you think that the best suitable approach is removing scripts those are already covered on pre-postconditions of unrelated test scenarios we can remove them. But my suggestion is separating all of them to be able to run each script independently. Otherwise, we should also remove login because we already use it in different scripts etc. 

As I said, it totally depends on the requirements.  Either way will work.